File: backend/src/main/java/com/falconenergy/service/DispatchService.java
package com.falconenergy.service;

import com.falconenergy.dto.DispatchResponse;
import com.falconenergy.dto.DispatchRequest;
import com.falconenergy.dto.LoadingActivityResponse;
import com.falconenergy.entity.LoadingActivity;
import com.falconenergy.entity.LoadingReport;
import java.util.List;

public interface DispatchService {
    List<LoadingActivityResponse> getPendingDispatchActivities();
    DispatchResponse createDispatch(Long loadingActivityId, DispatchRequest request);
    DispatchResponse createReadyDispatchForCompletedLoading(LoadingActivity activity, LoadingReport report);
    DispatchResponse getDispatchById(Long id);
    DispatchResponse getDispatchByActivityId(Long activityId);
    DispatchResponse releaseTruck(Long id);
    DispatchResponse startTransit(Long id);
}

File: src/services/dispatchService.js
import { api } from "./api";

export async function getPendingDispatchActivities() {
  return api.get("/dispatch/pending").then((r) => r.data);
}

export async function createDispatch(loadingActivityId, payload = {}) {
  return api.post(`/dispatch/create/${loadingActivityId}`, payload).then((r) => r.data);
}

export async function getDispatchById(id) {
  return api.get(`/dispatch/${id}`).then((r) => r.data);
}

export async function getDispatchByActivityId(activityId) {
  return api.get(`/dispatch/activity/${activityId}`).then((r) => r.data);
}

export async function releaseTruck(id) {
  return api.post(`/dispatch/${id}/release`).then((r) => r.data);
}

export async function cancelDispatch(id, payload = {}) {
  return api.post(`/dispatch/${id}/cancel`, payload).then((r) => r.data);
}

export async function startTransit(id) {
  return api.post(`/dispatch/${id}/start-transit`).then((r) => r.data);
}
